engine/src/exec-array.cpp

```diff
@@ -606,12 +606,23 @@ void MCArraysDoIntersect(MCExecContext& ctxt, MCArrayRef p_dst_array, MCArrayRef
 	MCNameRef t_key;
 	MCValueRef t_src_value;
     MCValueRef t_dst_value;
+    MCArrayRef t_orig_dst_array;
```

Comment:
Make this a MCAutoArrayRef t_orig_dst_array otherwise the reference won't be released.